torch/distributed/device_mesh.py

```diff
@@ -565,6 +566,13 @@ def init_device_mesh(
                     f"Found len(mesh_dim_names): {len(mesh_dim_names)} and len(mesh_shape):{len(mesh_shape)}.",
                 )
 
+        # assume valid device types are all letters
+        if device_type and not device_type.isalpha():
+            raise RuntimeError(
+                f"Device type with GPU index is not supported but got {device_type}. ",
+                "If you maintained a 'torch.device' object, it's recommended to pass in 'device.type'.",
```

Comment:
nit: we don't need the trailing `,`?

Reply:
you mean this? i think it's fine :D

![image](https://github.com/pytorch/pytorch/assets/49934421/b8309dcf-33a1-47d0-a434-969887aec24a)